tests/std/include/test_regex_support.hpp

```diff
@@ -116,6 +116,59 @@ class regex_fixture {
         }
     }
 
+    void should_capture_groups(const std::string& subject, const std::string& pattern,
+        const std::vector<std::string>& groups,
+        const std::regex_constants::syntax_option_type flags = std::regex_constants::ECMAScript) {
+        try {
+            const std::regex r(pattern, flags);
+            std::smatch m;
+
+            if (!std::regex_match(subject, m, r)) {
+                printf(R"(Expected regex("%s") to match "%s".)"
+                       "\n",
+                    pattern.c_str(), subject.c_str());
+                fail_regex();
+                return;
+            }
+
+            if (m[0] != subject) {
+                printf(R"(should_capture("%s", "%s"): not a full match)"
+                       "\nexpected: \"%s\""
+                       "\nfound:    \"%s\"\n",
+                    subject.c_str(), pattern.c_str(), subject.c_str(), m[0].str().c_str());
+                fail_regex();
+            }
+
+            if (m.size() != groups.size() + 1) {
+                printf(R"(should_capture("%s", "%s"): captures: %zd != %zd)"
+                       "\n",
+                    subject.c_str(), pattern.c_str(), m.size() - 1, groups.size());
+                fail_regex();
+            }
+
+            for (size_t i = 0; i < groups.size(); ++i) {
+                if (m[i + 1] != groups[i]) {
+                    printf(R"(should_capture("%s", "%s"): index %zd)"
+                           "\nexpected: \"%s\""
+                           "\nfound:    \"%s\"\n",
+                        subject.c_str(), pattern.c_str(), i, groups[i].c_str(), m[i + 1].str().c_str());
+                    fail_regex();
+                }
+            }
+        } catch (const std::regex_error& e) {
+            std::string groups_string = "|";
+            for (const auto& el : groups) {
+                groups_string.append(el);
+                groups_string.push_back('|');
+            }

```

Comment:
Suggestion: This seems confusing, because it doesn't look like the callsites. `|` means "alternations" for regexes, but here you're using it with an unofficial meaning to separate capture groups.

Would formatting the groups as `{"group1","group2","group3"}` be better?